backend/src/rss.rs

```diff
@@ -20,68 +22,110 @@ struct Event {
+    let db = db::get_conn_or_service_unavailable(db_pool).await?;
+
     let query = "select opencast_id, title, description from series where id = $1";
-    let series_data = db.query_one(query, &[&series_id]).await?;
+    let series_data = match db.query_one(query, &[&series_id]).await {
+        Ok(data) => data,
+        Err(_) => return Err(bad_request("DB error querying series data")),
+    };
+
     let series_oc_id = series_data.get::<_, String>("opencast_id");
-    let series_title = series_data.get::<_, String>("title");
+    let series_title = series_data.get::<_, String>("title").clone();
     let series_description = series_data.get::<_, Option<String>>("description");
     let cover_image = context.config.theme.logo.large.path.clone();
 
-    let video_items_result = generate_video_items(
-        &db,
-        &series_oc_id,
-        &series_title,
-        &rss_link,
-        &tobira_url,
-    );
+    // Build rss xml
+    let mut xml = XMLBuilder::new()
+        .version(XMLVersion::XML1_0)
+        .encoding("UTF-8".into())
+        .build();
+
+    let mut rss = XMLElement::new("rss");
+
+    // Add rss attributes
+    let attributes = [
+        ("version", "2.0"),
+        ("xmlns:dc", "http://purl.org/dc/elements/1.1/"),
+        ("xmlns:content", "http://purl.org/rss/1.0/modules/content/"),
+        ("xmlns:atom", "http://www.w3.org/2005/Atom"),
+        ("xmlns:media", "http://search.yahoo.com/mrss/"),
+        ("xmlns:itunes", "http://www.itunes.com/dtds/podcast-1.0.dtd"),
+    ];
+    for (name, value) in attributes {
+        rss.add_attribute(name, value);
+    }
+
+    let mut channel = XMLElement::new("channel");
+
+    // Add channel tags
+    let channel_data = [
+        ("title", vec![("", series_title.clone())]),
+        ("link", vec![("", series_link)]),
+        ("description", vec![("", series_description.unwrap_or_default())]),
+        ("language", vec![("", "und".to_string())]),
+        ("itunes:explicit",vec![("", "true".to_string())]),
+        ("itunes:image", vec![("href", (&cover_image.to_string_lossy()).to_string())]),
+        ("itunes:category", vec![("text", "Education".to_string())]),
+        ("atom:link", vec![
+            ("href", rss_link.clone()),
+            ("rel", "self".to_string()),
+            ("type", "application/rss+xml".to_string()),
+        ]),
+    ];
+    add_elements_to_xml(&channel_data, &mut channel);
+
+    // Add video items
+    let video_items = match generate_video_items(
+        &db, &series_oc_id, &series_title, &rss_link, &tobira_url).await {
+        Ok(items) => items,
+        Err(_) => return Err(bad_request("empty series")),
+    };
+
+    for item in video_items {
+        // I have no idea how to correctly handle items that are resulting in an error.
+        // Pretty sure however that this here is a bad solution and would result in
+        // potentially broken feeds.
+        let _ = channel.add_child(item);
+    }
```

Comment:
What in the world am I doing with this? Nothing else seems to work but this surely can't be the solution.

Reply:
Ok I just needed to unwrap this, whoops.

Reply:
Yep exactly